### DISCUSSION TITLE
Use QueryClient in vanilla js, outside React?

### DISCUSSION THREAD

**Discussion by cambiata** [TanStack/query#2805] [Q&A]
I have several projects where I need to manage application (server) state but where this is fully or partly separated from React UI. Is it possible to interact with the QueryClient in such a way that I can use it as a data layer outside React?
Specifically:
- add, run and invalidate queries 
- add listeners to act upon specific data changes, errors, loading states etc 

Or are there other vanilla js data layers out there suitable for my non-React needs, that can be tweeked with custom hooks to work well in a React context when that's needed.?

**Comment by TkDodo** [comment]
The react-query core is framework agnostic, and people have ported it to vue-query and svelte-query already, so yes, I'd definitely say you can.
